src/App.jsx

```diff
@@ -14,70 +16,68 @@ export const goods = [
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const isGoodSelected = good => good === selectedGood;
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  const handleGoodClick = (good) => {
+    setSelectedGood(!isGoodSelected(good) ? good : '');
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  return (
+    <main className="section container">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      {!selectedGood.length ? (
+        <h1 className="title is-flex is-align-items-center">
+          No goods selected
+        </h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood} is selected`}
+          <button
+            onClick={() => {
+              setSelectedGood('');
+            }}
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3 RemoveButton"
+          />
+        </h1>
+      )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        {goods.map(good => (
+          <tbody key={good}>
+            <tr
+              data-cy="Good"
+              className={classnames({
+                'has-background-success-light': isGoodSelected(good),
+              })}
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              <td>
+                <button
+                  onClick={() => {
+                    handleGoodClick(good);
+                  }}
+                  data-cy={
+                    isGoodSelected(good) ? 'RemoveButton' : 'AddButton'
+                  }
+                  type="button"
+                  className={classnames('button', {
+                    'is-info': isGoodSelected(good),
+                  })}
+                >
+                  {isGoodSelected(good) ? '-' : '+'}
+                </button>
```

Comment:
if you change almost everything depending on the condition isGoodSelected(good) 
it will be better to rewrite it like this
```
{isGoodSelected(good)
  ? <button></button>
  : <button></button>
}
```